src/View/Requirements_Backend.php

```diff
@@ -1421,6 +1440,32 @@ function () use ($fileList, $minify, $type) {
         return $combinedURL;
     }
 
+    /**
+     * Resolves relative paths in CSS files which are lost when combining them
+     *
+     * @param string $content
+     * @param string $filePath
+     * @return string New content with paths resolved
+     */
+    protected function resolveCSSReferences($content, $filePath)
+    {
+        $doResolving = Config::inst()->get(__CLASS__, 'resolve_relative_css_refs');
+        if (!$doResolving) {
+            return $content;
+        }
+        $fileUrl = Injector::inst()->get(ResourceURLGenerator::class)->urlForResource($filePath);
+        $fileUrlDir = dirname($fileUrl);
+        $content = preg_replace_callback('#(url\([\n\r\s\'"]*)([^\s\)\?\'"]+)#i', function ($match) use ($fileUrlDir) {
+            [ $_, $prefix, $relativePath ] = $match;
+            if ($relativePath[0] === '/' || false !== strpos($relativePath, '://')) {
+                return $prefix . $relativePath;
+            }
+            $full = FilesystemPath::canonicalize($fileUrlDir . '/' . $relativePath);
```

Comment:
How far up does this let URIs escape? Since we're doing this to allow themes to escape into other themes, I just want to have an understanding of how far exactly they're able to go.

Reply:
They are able to go down to the document root - just like any other css file.
i.e. in case we work with
`https://example.com/themes/themeONE/folder1/folder2/style.css`
`canonicalize()` will be given paths like `themes/themeONE/folder1/folder2/../../something.jpg`

Reply:
In that example they'd still be inside the `themeONE` folder but yeah that's fine. I guess it's not giving theme css access to any files they can't already access.

Reply:
No, we're not reading any files.
Also, test marked with 
`/* COMBINED CSS URL RESOLVER DECODED 4 DOUBLE-DOT WHEN ONLY 3 LEVELS AVAILABLE*/`
shows exactly what's going to happen when we try to use more '../' than available.